<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>LspConfigWidget</class>
 <widget class="QWidget" name="LspConfigWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>671</width>
    <height>770</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>Client Settings</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <item>
        <layout class="QFormLayout" name="formLayout">
         <property name="formAlignment">
          <set>Qt::AlignmentFlag::AlignHCenter|Qt::AlignmentFlag::AlignTop</set>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item row="0" column="1">
          <widget class="QCheckBox" name="chkOnTypeFormatting">
           <property name="text">
            <string>Format on typing</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QCheckBox" name="chkFmtOnSave">
           <property name="text">
            <string>Format on save</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QCheckBox" name="chkSemanticHighlighting">
           <property name="text">
            <string>Enable semantic highlighting</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QCheckBox" name="chkInlayHint">
           <property name="text">
            <string>Enable inlay hints</string>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Completions:</string>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QCheckBox" name="chkShowCompl">
           <property name="text">
            <string>Show completions</string>
           </property>
          </widget>
         </item>
         <item row="6" column="1">
          <widget class="QCheckBox" name="chkComplDoc">
           <property name="text">
            <string>Show inline docs for selected completion</string>
           </property>
          </widget>
         </item>
         <item row="7" column="1">
          <widget class="QCheckBox" name="chkSignatureHelp">
           <property name="text">
            <string>Show function signature when typing a function call</string>
           </property>
          </widget>
         </item>
         <item row="8" column="1">
          <widget class="QCheckBox" name="chkComplParens">
           <property name="text">
            <string>Add parentheses upon function completion</string>
           </property>
          </widget>
         </item>
         <item row="9" column="1">
          <widget class="QCheckBox" name="chkAutoImport">
           <property name="text">
            <string>Add imports automatically if needed upon completion</string>
           </property>
          </widget>
         </item>
         <item row="10" column="0">
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>Navigation:</string>
           </property>
          </widget>
         </item>
         <item row="10" column="1">
          <widget class="QCheckBox" name="chkRefDeclaration">
           <property name="text">
            <string>Count declarations when searching for references to a symbol</string>
           </property>
          </widget>
         </item>
         <item row="11" column="1">
          <widget class="QCheckBox" name="chkAutoHover">
           <property name="text">
            <string>Show information about currently hovered symbol</string>
           </property>
          </widget>
         </item>
         <item row="12" column="1">
          <widget class="QCheckBox" name="chkHighlightGoto">
           <property name="text">
            <string>Highlight target line when hopping to it</string>
           </property>
          </widget>
         </item>
         <item row="14" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Server:</string>
           </property>
          </widget>
         </item>
         <item row="14" column="1">
          <widget class="QCheckBox" name="chkDiagnostics">
           <property name="text">
            <string>Show program diagnostics</string>
           </property>
          </widget>
         </item>
         <item row="15" column="1">
          <widget class="QCheckBox" name="chkMessages">
           <property name="text">
            <string>Show notifications from the LSP server</string>
           </property>
          </widget>
         </item>
         <item row="16" column="1">
          <widget class="QCheckBox" name="chkIncrementalSync">
           <property name="text">
            <string>Incrementally synchronize documents with the LSP server</string>
           </property>
          </widget>
         </item>
         <item row="17" column="0">
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>Document outline:</string>
           </property>
          </widget>
         </item>
         <item row="17" column="1">
          <widget class="QCheckBox" name="chkSymbolSort">
           <property name="text">
            <string>Sort symbols alphabetically</string>
           </property>
          </widget>
         </item>
         <item row="18" column="1">
          <widget class="QCheckBox" name="chkSymbolDetails">
           <property name="text">
            <string>Display additional details for symbols</string>
           </property>
          </widget>
         </item>
         <item row="19" column="1">
          <widget class="QCheckBox" name="chkSymbolTree">
           <property name="text">
            <string>Present symbols in a hierarchy instead of a flat list</string>
           </property>
          </widget>
         </item>
         <item row="20" column="1">
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <property name="leftMargin">
            <number>20</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QCheckBox" name="chkSymbolExpand">
             <property name="text">
              <string>Automatically expand tree</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="13" column="1">
          <widget class="QCheckBox" name="chkHighlightSymbol">
           <property name="text">
            <string>Highlight symbol under cursor</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Orientation::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_4">
      <attribute name="title">
       <string>Allowed &amp;&amp; Blocked Servers</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QListWidget" name="allowedAndBlockedServers"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>User Server Settings</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Settings File:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="KUrlRequester" name="edtConfigPath" native="true">
           <property name="text" stdset="0">
            <string/>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QPlainTextEdit" name="userConfig"/>
       </item>
       <item>
        <widget class="QLabel" name="userConfigError">
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_3">
      <attribute name="title">
       <string>Default Server Settings</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_9">
       <item>
        <widget class="QPlainTextEdit" name="defaultConfig">
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KUrlRequester</class>
   <extends>QWidget</extends>
   <header>kurlrequester.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
